File: Problems/No.724_Find_Pivot_Index/find_pivot_index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        
        // parameters
        int num_sum = 0;
        int left_sum = 0;
        int pivot_index = 0;

        // start computing
        for (auto &num: nums) num_sum += num;
        for (auto &num: nums) {

            num_sum -= num;
            if (num_sum == left_sum) return pivot_index;
            else pivot_index++;
            left_sum += num;

        }

        return -1;

    }
};

File: Problems/No.724_Find_Pivot_Index/find_pivot_index.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        # parameters
        num_sum, left_sum = 0, 0
        
        # start computing
        for num in nums:
            num_sum += num
        for pivot_idx, num in enumerate(nums):
            num_sum -= num
            if (num_sum == left_sum):
                return pivot_idx
            else:
                left_sum += num
        
        return -1
